sermant-plugins/sermant-springboot-registry/springboot-registry-service/src/main/java/io/sermant/discovery/service/lb/discovery/nacos/NacosDiscoveryClient.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.discovery.service.lb.discovery.nacos;
+
+import com.alibaba.nacos.api.exception.NacosException;
+import com.alibaba.nacos.api.naming.NamingService;
+import com.alibaba.nacos.api.naming.pojo.Instance;
+import com.alibaba.nacos.api.naming.pojo.ListView;
+
+import io.sermant.core.common.LoggerFactory;
+import io.sermant.core.plugin.config.PluginConfigManager;
+import io.sermant.core.utils.StringUtils;
+import io.sermant.discovery.config.NacosRegisterConfig;
+import io.sermant.discovery.config.RegisterType;
+import io.sermant.discovery.entity.RegisterContext;
+import io.sermant.discovery.entity.ServiceInstance;
+import io.sermant.discovery.service.lb.discovery.ServiceDiscoveryClient;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * NACOS connects to the client
+ *
+ * @author xiaozhao
+ * @since 2024-11-16
+ */
+public class NacosDiscoveryClient implements ServiceDiscoveryClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    private final NacosServiceManager nacosServiceManager = NacosServiceManager.getInstance();
+
+    private Instance instance;
+
+    private final NacosRegisterConfig nacosRegisterConfig;
+
+    /**
+     * constructor
+     */
+    public NacosDiscoveryClient() {
+        nacosRegisterConfig = PluginConfigManager.getPluginConfig(NacosRegisterConfig.class);
+    }
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public boolean registry(ServiceInstance serviceInstance) {
+        String serviceId = serviceInstance.getServiceName();
+        String group = nacosRegisterConfig.getGroup();
+        instance = nacosServiceManager.buildNacosInstanceFromRegistration(serviceInstance);
```

Comment:
Should we consider the case where the host application has already been using the Nacos client?